File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
       int m = word1.length(), n= word2.length();
       StringBuilder result = new StringBuilder();
       if(word1 != null || word2!= null){
       for(int i=0; i<Math.max(m,n); i++){
        if(i<m){
            result.append(word1.charAt(i));
        }
        if(i<n){
            result.append(word2.charAt(i));
        }
       } 
        return result.toString();

       } else {
        if(word1!= null){
            return word1;
        } else {
            return word2;
        }
       }
    }
}

File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1:str, word2:str) -> str:
        l = len(word1)
        m = len(word2)
        n = max(l,m)
        output = []

        for i in range(n):
            if i < l:
                output += word1[i]
            if i < m:
                output +=word2[i]
        return "".join(output)
